// Copyright (c) 2015 Arista Networks, Inc.  All rights reserved.
// Arista Networks, Inc. Confidential and Proprietary.
// Subject to Arista Networks, Inc.'s EULA.
// FOR INTERNAL USE ONLY. NOT FOR DISTRIBUTION.

syntax = "proto3";

import "google/protobuf/timestamp.proto";

option go_package = "gen";

message Notification {
    google.protobuf.Timestamp timestamp = 1; // Timestamp in nanoseconds resolution.
    string path = 2;  // Deprecated
    bool delete_all = 3;
    repeated bytes deletes = 4; // BDSN encoded
    message Update {
        bytes key = 1; // BDSN encoded
        bytes value = 2; // BDSN encoded
    }
    repeated Update updates = 5;
    repeated bytes retracts = 6; // BDSN encoded
    repeated bytes path_elements = 7; // BDSN encoded
}

message Dataset {
   string type = 1;
   string name = 2;
}

message NotificationBatch {
    // TODO: deprecate, for now just assume
    // that type is "device"
    string d = 1;
    repeated Notification notifications = 2;
    Dataset dataset = 3;
}
